cpp/src/parquet/file_writer.cc

```diff
@@ -356,6 +355,15 @@ class FileSerializer : public ParquetFileWriter::Contents {
 
   RowGroupWriter* AppendBufferedRowGroup() override { return AppendRowGroup(true); }
 
+  void AddKeyValueMetadata(
+      std::shared_ptr<const KeyValueMetadata> key_value_metadata) override {
```

Comment:
Should it better be a `const std::shared_ptr<const KeyValueMetadata>& key_value_metadata`? Because it doesn't force copy everytime

Reply:
Fixed